### DISCUSSION TITLE
Reproducibility Package Plan

### DISCUSSION THREAD

**Discussion by umarbilgrammi** [ecn310/course-project-taxes-tariffs#20] [Announcements]
:)

**Comment by DylanTh20** [comment]
Raw Data 
The data sets can be acquired through from the World Integrated Trade Solution website. To get the
data, we altered the indicator on the website allowing us to get our two different data sets. Specifically,
we used the indicators Taxes on Goods and Services (percent of revenue) and Taxes on international
trade (percent of revenue). We downloaded these datasets by clicking the download button on the
table view tab located on the right hand side above the data

**Comment by umarbilgrammi** [comment]
Task 1: 

- [ ] Raw data 
  - Instructions to download and find the data set --> a lot of detail 
  - summary statistics
  - use codebook on all variables
  - read me on github
  - master do file explaining the whole process 
  - delete unnecessary files / logs

**Comment by umarbilgrammi** [comment]
Task 2 - Raw data to research outputs 

- [ ] All code needed to go from beginning data set to merged.dta
  - read me encompassing the whole thing especially for our intermediate data sets 
  - five location on how to access all files including the figures we created 
  - make a to do list of how many do-files there are 
  - include any necessary log files

**Comment by umarbilgrammi** [comment]
Task 3 - 1 log file for each do file 
- [ ]  any few pieces we are not directly exporting --> make reproducible by adding log file for things that are not directly exported

**Comment by Ksarrge** [comment]
## Reproducibility Package Plan

| Task | Target Date | Assignee | Notes |
|---|---|---|---|
| Log Files | Nov. 22 | Umar | Create a log file for each do file |
| Sweep GitHub | Nov. 22 | Kathryn | Delete unnecessary files/logs and ensure all our files within repo are organized and labeled effectively |
| Instructions to Download Raw Data | Nov. 24 | Dylan | Write a step by step of how to download and access the VAT and International taxes and tariffs datasets |
| Create Master Do. file | Dec. 1 | Umar | Finish all code needed to reproduce our results |
| Check Do. file | Dec. 3 | Kathryn | Run Master Do. file to check if the reproducibility operates successfully |
| Fix and Add any Notes to Do. File | Dec. 5 | Dylan | If needed, fix any and all of the mistakes in the Do. files |
| Write README | Dec. 5 | Umar, Kathryn, and Dylan | Write a high-level documentation on the use and structure of the code that should explain any intermediate datasets and give location and how to access all files, including figures which we will do in GitHub |
| Documentation test by insider | Dec. 7 | Kathryn | Test our documentation to ensure that someone else could find and open everything they would need to reproduce our work |
| Data Appendix | Dec.5 | Umar, Kathryn, and Dylan | Write a report about how exactly we did our research process |
| Documentation test by outsider/peer | Dec. 10 | Dylan's friend | Have an outsider test our documentation to ensure they can find and open everything they would need to reproduce our work |
| Fix Documentation | Dec. 14 | Umar | Fix any and all mistakes in the documentation |
| Final Documentation | Dec. 16 | Umar, Kathryn, and Dylan | Read through the documentation and double check everything looks how it should |

**Reply by kbuzard** [reply]
@ecn310/taxes-tariffs This looks like a solid plan. One thing I think will need to be clarified: where is everything going to live? That is, right now you don't have any subfolders (except one: more on that below). I think you'll need one at the very least--for all your figures. You may need more depending on how many do-files, logs, etc. you need.
- The subfolder *.github*: I think that Overleaf creates this if you upload files directly into Overleaf. Do you remember if you uploaded [mergedhistogram.png](https://github.com/ecn310/course-project-taxes-tariffs/blob/main/.github/mergedhistogram.png) directly into Overleaf?
  - You'll want to create a folder for outputs (you can call it "figures_tables" or whatever you want that makes sense) and then put all your figures in there to pull into Overleaf. Then delete this .github folder and change the reference in your report to wherever the new figure is stored.
